### PR TITLE
ROX-16608: Normalize process and process listening on ports to utf8

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Ensure that elements from processes and processes listening on ports are valid UTF8 strings. Example of a failure. nulls are also not valid in Postgres UTF8 so cover that case too

log
```
detection/lifecycle: 2023/04/19 22:40:05.604034 manager_impl.go:198: Error: Error adding process indicators: ERROR: invalid byte sequence for encoding "UTF8": 0xff (SQLSTATE 22021)
```
due to
```
ls $(iconv -f UTF-8 -t UTF-16 <(echo 'abc#$%%^'))
```

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Will re-run the above command
Screenshot 2023-04-19 at 5 02 15 PM

### CODE REVIEW

**Review by dashrews78 — APPROVED** [review]
LGTM.  Add the milestone for `4.0.0-rc.6` and you may want to get with the 3.74 folks.

File: migrator/migrations/n_42_to_n_43_postgres_process_indicators/migration.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ func move(gormDB *gorm.DB, postgresDB *postgres.DB, legacyStore legacy.Store) er
 	pgutils.CreateTableFromModel(context.Background(), gormDB, frozenSchema.CreateTableProcessIndicatorsStmt)
 	var processIndicators []*storage.ProcessIndicator
 	err := walk(ctx, legacyStore, func(obj *storage.ProcessIndicator) error {
+		normalize.NormalizeIndicator(obj)
```
Comment (c-du) [inline migrator/migrations/n_42_to_n_43_postgres_process_indicators/migration.go:51]: Not used yet?

**Review by c-du — APPROVED** [review]

File: pkg/process/normalize/normalize.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+// NormalizeIndicator ensures that that indicator will comply with UTF8 encoding
+func NormalizeIndicator(indicator *storage.ProcessIndicator) {
+	signal := indicator.GetSignal()
+	if signal == nil {
+		return
+	}
+	signal.ExecFilePath = sanitizeString(signal.GetExecFilePath())
+	signal.Name = sanitizeString(signal.GetName())
+	signal.Args = sanitizeString(signal.GetArgs())
```
Comment (robbycochran) [inline pkg/process/normalize/normalize.go:23]: Also, need to sanitize `ParentExecFilePath` strings in `LineageInfo`

Comment (connorgorman) [reply pkg/process/normalize/normalize.go:23]: thx

File: migrator/migrations/n_42_to_n_43_postgres_process_indicators/migration.go (hunk 1 above)
Comment (connorgorman) [reply migrator/migrations/n_42_to_n_43_postgres_process_indicators/migration.go:51]: hm?

File: pkg/process/normalize/normalize.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+// NormalizeNetworkEndpoint normalizes network endpoints
+func NormalizeNetworkEndpoint(endpoint *sensor.NetworkEndpoint) {
```
Comment (robbycochran) [inline pkg/process/normalize/normalize.go:27]: Use this function in `sensor/common/networkflow/manager/manager_impl.go`

File: pkg/process/normalize/normalize.go (hunk 2 above)
Comment (connorgorman) [reply pkg/process/normalize/normalize.go:23]: done

File: pkg/process/normalize/normalize.go (hunk 3 above)
Comment (connorgorman) [reply pkg/process/normalize/normalize.go:30]: just added that, thanks

**Review by erthalion — APPROVED** [review]
Sounds reasonable, with the caveat that we need to be sure there is no genuine encoding mismatch between client and the db. Otherwise the data will be silently modified -- although I hope this will trigger some alerting.

**Comment by connorgorman** [comment]
> Sounds reasonable, with the caveat that we need to be sure there is no genuine encoding mismatch between client and the db. Otherwise the data will be silently modified -- although I hope this will trigger some alerting.

@erthalion Good point, the default has been UTF8 in every run we've had, but I submitted a PR so that initdb explicitly specifies `--encoding`

**Pull request merged**